mcproto/types/identifier.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+import re
+from typing_extensions import override
+
+from mcproto.buffer import Buffer
+from mcproto.types.abc import MCType
+from attrs import define
+from mcproto.types.nbt import NBTag, StringNBT, NBTagConvertible
+
+
+@define
+class Identifier(MCType, NBTagConvertible):
+    """A Minecraft identifier.
+
+    :param namespace: The namespace of the identifier.
+    :param path: The path of the identifier.
+    :type path: str, optional
+
+    If the path is not provided, the namespace and path will be extracted from the :attr:`namespace` argument.
+    If the namespace is not provided, it will default to "minecraft".
+    """
+
+    namespace: str
+    path: str
+
+    @override
+    def __init__(self, namespace: str, path: str | None = None) -> None:
+        if namespace.startswith("#"):
+            namespace = namespace[1:]
+
+        if path is None:
+            if ":" not in namespace:
+                namespace, path = "minecraft", namespace  # Default namespace
+            else:
+                namespace, path = namespace.split(":", 1)
```

Comment:
The converters from attrs would probably be a better choice than overriding `__init__` here.

Reply:
I think this particular example should do without `attrs`'s `__init__`.

I tried to use the converters and validators but when a variable depends on another I don't think it is possible to write things in a way that is supported by attrs. It is much simpler and more legible to just use a simple `__init__` method which does what it's supposed to do nicely.

Reply:
Ah I see, yeah that makes sense, keep the init then, but do mention why in a comment there somewhere